python/cuml/test/test_kmeans.py

```diff
@@ -109,6 +109,45 @@ def test_kmeans_sequential_plus_plus_init(nrows, ncols, nclusters,
     assert abs(cu_score - sk_score) <= cluster_std * 1.5
 
 
+@pytest.mark.parametrize('nrows', [1000, 10000])
+@pytest.mark.parametrize('ncols', [10, 50])
+@pytest.mark.parametrize('nclusters', [5, 10])
+@pytest.mark.parametrize('max_weight', [10, 20])
+@pytest.mark.parametrize('random_state', [i for i in range(5)])
+def test_weighted_kmeans(nrows, ncols, nclusters,
+                         max_weight, random_state):
+
+    # Using fairly high variance between points in clusters
+    cluster_std = 1.0
+    np.random.seed(random_state)
+
+    # set weight per sample to be from 1 to max_weight
+    wt = np.random.randint(1, high=max_weight, size=nrows)
+
```

Comment:
What if a user passes in a weight less than zero? Exception? Or weird result?

Reply:
We would see weird results @JohnZed. It looks like sklearn behaves strangely too. I can enforce positive sample weights with a check at C++ level OR do we want this at python level?